File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ans;
        sort(intervals.begin(), intervals.end());
        vector<int> temp_interval;
        temp_interval = intervals[0];

        for(auto it : intervals){
            if(it[0] <= temp_interval[1]){
                temp_interval[1] = max(temp_interval[1], it[1]);
            }else{
                ans.push_back(temp_interval);
                temp_interval = it;
            }
        }
        ans.push_back(temp_interval);
        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        temp_interval = intervals[0]
        ans = []
        for interval in intervals:
            if temp_interval[1] >= interval[0]:
                temp_interval[1] = max(temp_interval[1], interval[1])
            else:
                ans.append(temp_interval)
                temp_interval = interval
        ans.append(temp_interval)
        return ans